File: Example 13.go
package main

import (
	"fmt"
	"strconv"
)

func generateData(n int) []string {
	baseNumber := "082"
	var customers []string
	var mobileNumber string

	for i := 0; i < n; i++ {
		mobileNumber = baseNumber + strconv.Itoa(i+1)
		for len(mobileNumber) < 12 {
			mobileNumber = "0" + mobileNumber
		}
		customers = append(customers, mobileNumber)
	}
	return customers
}

func sendPromoDiscount(input1 []string, input2 []string) {
	for i := 0; i < len(input1); i++ {
		fmt.Println("Sending Promo to " + input1[i])
	}
	fmt.Println("Its Finished to send Promo to " + strconv.Itoa(len(input1)) + " Customers")

	for i := 0; i < len(input2); i++ {
		fmt.Println("Sending Discount to " + input2[i])
	}
	fmt.Println("Its Finished to send Discount to " + strconv.Itoa(len(input2)) + " Customers")
}

func main() {
	dataPromo := generateData(100000)
	dataDiscount := generateData(1000)

	sendPromoDiscount(dataPromo, dataDiscount)
}

File: Example 13.js
function generateData(n){
    const baseNumber = '082' ;
    let customers = [];
    let mobileNumber;

    for (let i = 0 ; i < n; i++){
        mobileNumber = baseNumber + (String(i).padStart(9, '0'));
        customers.push(mobileNumber)
    }
    return customers;
}

function sendPromoDiscount(input1, input2){
    for(let i = 0; i < input1.length; i++){
        console.log('Sending Promo to ' + input1[i]);
    }
    console.log('Its Finished to send Promo to ' + input1.length + 'Customers');

    for (let i = 0; i < input2.length; i++){
        console.log('Sending Discount to ' + input2[i]);
    }
    console.log('Its Finished to send Discount to ' + input2.length + ' Customers');
}

let dataPromo = generateData(100000);
let dataDiscount = generateData(1000);

sendPromoDiscount(dataPromo, dataDiscount);

